src/components/ExerciseSearchBar.tsx

```diff
@@ -57,64 +71,88 @@ const ExerciseSearchBar: React.FC<ExerciseSearchBarProps> = ({ onSelectExercise
-                <View>
-                  <Text className="text-l mx-1 my-3 font-light">Recently Used</Text>
-                  <TouchableOpacity
-                    accessibilityRole="button"
-                    key={'recently-used-' + recentlyUsedExercise.current}
-                    onPress={() => onPressHandler(recentlyUsedExercise.current)}
-                    className="mb-1 rounded border border-slate-200 p-3"
-                  >
-                    <Text className="text-l font-bold ">{recentlyUsedExercise.current}</Text>
-                  </TouchableOpacity>
-                </View>
-              ) : (
-                <View />
-              )}
-              <Text className="text-l mx-1 my-3 font-light ">All Exercises</Text>
-              {allExercises.map((item) => {
-                return (
-                  <TouchableOpacity
-                    accessibilityRole="button"
-                    key={item}
-                    onPress={() => onPressHandler(item)}
-                    className="mb-1 rounded border border-slate-200 p-3"
-                  >
-                    <Text className="text-l font-bold">{item}</Text>
-                  </TouchableOpacity>
-                );
-              })}
-            </View>
-          ) : (
-            <View className="mx-5 ">
-              <Text className="my-1 text-xs font-light "> Search Result:</Text>
-              {searchArray.current.map((item) => {
-                return (
-                  <TouchableOpacity
-                    accessibilityRole="button"
-                    key={item}
-                    onPress={() => onPressHandler(item)}
-                    className="mb-1 rounded border border-slate-200 p-3"
-                  >
-                    <Text className="text-l font-bold ">{item}</Text>
-                  </TouchableOpacity>
-                );
-              })}
-            </View>
-          )}
-        </ScrollView>
-      </View>
+      {hideBodyOnIdle && !searchOnFocus ? (
+        <View className="h-1 w-11/12 rounded border-t border-slate-200" />
+      ) : (
+        <View className="divide-y-10 absolute mt-20 h-60 w-11/12 overflow-scroll rounded border border-slate-200">
+          <ScrollView keyboardShouldPersistTaps="always">
+            {searchInput.length <= 0 ? (
+              <View className="mx-5 ">
+                {filters.length > 0 ? (
+                  <View className="border-b-0.5 flex flex-row border-slate-400 pb-2">
+                    <ScrollView horizontal={true}>
+                      {filters.map((item) => {
+                        return <Chip key={item} value={item} closeable={false} />;
+                      })}
+                      {/**END OF CHIP*/}
+                    </ScrollView>
+                  </View>
+                ) : (
+                  <View />
+                )}
+                <Text className="text-l mx-1 my-3 font-light ">All Exercises</Text>
+                {allExercises
```

Comment:
This doesn't have to be fixed this PR, but i'm starting to get warnings about the number of touchable opacity elements being rendered. There might be too many exercises now where its not performant to render everything all at once. Possible solutions: don't display all, only display search results OR implement pagination.

Reply:
hmm, lets see if thats gonna be a major issue later on.  I didn't get those warnings.   If it keeps showing up on a lot of devices maybe more fetches can be done